### ISSUE TITLE
Support hapi v21

### ISSUE THREAD

**Issue by blumamir** [open-telemetry/opentelemetry-js-contrib#1383]
hapi instrumentation has [support for version range '>=17 <21', as fixed in #1382 

The package released a new major version [21.0.0](https://www.npmjs.com/package/@hapi/hapi/v/21.0.0) at 11/07/2022, and we should update the instrumentation accordingly:

supported version for patch
README
unit tests
test-all-versions (currently missing).

Relevant renovate PR, with the output errors of trying to bump the instrumentation version to 22: #1280 

Looks like v21 now publishes it's own types and they are not compatible with previous versions of hapi. Not sure how to handle this in instrumentation

**Comment by toddtarsi** [comment]
Yes, I'd like to help here if possible. This installs the legacy definitely typed package, which overrides my built in types and breaks all sorts of other places in monorepos because those type definitions override the modern builtins.

**Comment by blumamir** [comment]
Sure, thanks 👍

**Comment by Towerism** [comment]
My team is waiting on this. Do you need help @toddtarsi?

**Comment by toddtarsi** [comment]
@blumamir @Towerism - Uh oh. I'm sorry, I just saw this was assigned to me.  Sometimes my github notifications get a bit overwhelming. I'd be happy to start taking a look tonight. I believe we'll just change the dependency in this file to just drop the types package and bring up hapi to v21, and glue up wherever the types complain. If not me, thats fine. Sorry about that all!

https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-hapi/package.json

**Comment by toddtarsi** [comment]
@Towerism @blumamir - Okay, I made a PR for this, but I will need help testing it and such.

**Comment by Boelensman1** [comment]
@toddtarsi PR was merged, this issue can be closed
